### ISSUE TITLE
xtext.tests says it bundles lib/simple.jar but it does not seem to be in jar

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-core#2035]
Bundle-ClassPath: lib/simple.jar,.

**Comment by cdietrich** [comment]
https://ci.eclipse.org/xtext/job/xtext-eclipse/job/cd_xtext_issue2125_mwe/1/console

**Comment by LorenzoBettini** [comment]
That file is giving me headaches in https://github.com/eclipse/xtext-core/issues/2052

The two tests:
- org.eclipse.xtext.resource.ClassloaderClasspathUriResolverTest.testClasspathUriForFileInJar()
- org.eclipse.xtext.resource.ClassloaderClasspathUriResolverTest.testClasspathUriForFileInJarWithFragment()
fail because they can't find the simple.ecore stored in the JAR.
I need to make "lib" a source folder, but still that test case must be run as a plug-in unit test to make it work, but that sounds wrong to me.

**Comment by cdietrich** [comment]
can you check if arne found a solution last time
https://github.com/eclipse/xtext-core/tree/ade-newbuild
am also not sure about eclipse tests vs pure surefire tests

**Comment by LorenzoBettini** [comment]
Actually, I found a solution for running that as a pure JUnit test from Eclipse: no need to set lib as a source folder, as long as you add the JAR in the launch:

![gnome-shell-screenshot-duao1p](https://user-images.githubusercontent.com/1202254/212662491-5dcad2c3-f02e-4c1c-8eba-f650a2e07059.png)

I guess those tests must be run as pure JUnit tests: many other tests in that project fail if run as plug-in tests.

**Comment by cdietrich** [comment]
yes. but if we put the jar to the build.properties and manifest.
will it be on the classpath?

**Comment by LorenzoBettini** [comment]
Only if we run that as a plug-in test I guess, at least when running in Eclipse.
With the launch configuration as above it works from Eclipse.
I have to check how it works from Maven when executed with maven-surefire (NOT tycho-surefire).

**Comment by LorenzoBettini** [comment]
update: it works like a charm with maven-surefire :)
it looks like not all the same tests are run by maven-surefire: Eclipse runs more of them.
As it happened in .ide.tests, it's just that some test classes don't follow the convention for names, I'll check that.
However, in general, in that project only JUnit4 tests must be executed, right?

**Comment by LorenzoBettini** [comment]
But in that project, we don't need that JAR for tests... or did you mean something else?

**Comment by cdietrich** [comment]
yes this is what we want

**Comment by LorenzoBettini** [comment]
All 12 projects build and all tests are green! :D

**Comment by cdietrich** [comment]
nice

**Comment by LorenzoBettini** [comment]
@cdietrich @szarnekow everything is working in Eclipse and in the CI (GitHub Actions).
I'd be ready to push the branch of my fork as a branch in the main Git repository to adjust the Jenkins build.
Before doing that, I'd like to know whether it's OK for you to leave the GitHub Actions in place. The benefits of GitHub Actions is that we can test also on macOS and Windows (it's already like that in my fork, and all the builds are green)

**Comment by cdietrich** [comment]
@LorenzoBettini  does this also produce a local maven repo (aka publishing) yet.

**Comment by LorenzoBettini** [comment]
That's another thing I have to try (first locally): I don't remember the syntax for deploying to a local repository and I'm still investigating.

For the moment it creates a p2 repository (I created a new project for that) in the same location as it was before.

**Comment by LorenzoBettini** [comment]
@cdietrich I think I got it.
Can I leave the GitHub Actions anyway?

**Comment by cdietrich** [comment]
i dont know if this will work or something needs to be enabled by admins

**Comment by cdietrich** [comment]
we can keep it for testing for now and later decide, am not sure how terrible eclipses quotas are

**Comment by LorenzoBettini** [comment]
I've been using GitHub Actions for EMF Parsley last month, intensively during the new simrel, and it worked perfectly :)

**Comment by LorenzoBettini** [comment]
OK, I'm ready to test it on Jenkinsfile, I'm going to push my branch on the main repo and let's see if I updated Jenkinsfile correctly...

**Comment by LorenzoBettini** [comment]
GitHub Actions are running in the main xtext-core repository...

**Comment by cdietrich** [comment]
with no more gradle this is obsolete
